elide-async/src/main/java/com/yahoo/elide/async/service/DefaultAsyncQueryDAO.java

```diff
@@ -87,14 +109,14 @@ public AsyncQuery updateAsyncQuery(UUID asyncQueryId, UpdateQuery updateFunction
     }
 
     @Override
-    public Iterable<Object> deleteAsyncQueryAndResultCollection(Iterable<Object> asyncQueryList) {
+    public Collection<AsyncQuery> deleteAsyncQueryAndResultCollection(Collection<AsyncQuery> asyncQueryList) {
         log.debug("deleteAsyncQueryAndResultCollection");
         executeInTransaction(dataStore, (tx, scope) -> {
             EntityProjection asyncQueryCollection = EntityProjection.builder()
                     .type(AsyncQuery.class)
                     .build();
 
-            Iterator<Object> itr = asyncQueryList.iterator();
+            Iterator<AsyncQuery> itr = asyncQueryList.iterator();
```

Comment:
Same comment about the load below.  The objects can be assumed to have already been loaded.  This function can just call tx.delete.

Reply:
We are able to remove the load object logic for the update, but does not work with delete. Even calling save before delete does not work. It throws TransactionException trying to remove a detached entity.